pandas/core/groupby.py

```diff
@@ -1873,33 +1922,81 @@ def cummax(self, axis=0, **kwargs):
 
         return self._cython_transform('cummax', numeric_only=False)
 
-    def _get_cythonized_result(self, how, grouper, needs_mask=False,
-                               needs_ngroups=False, **kwargs):
+    def _get_cythonized_result(self, how, grouper, aggregate=False,
```

Comment:
Had to add some new keywords to make this more dynamic and support aggregation in addition to the transformations it was already doing with `shift` and `fillna`. I think most could be generally applicable save `result_is_index` which may be specific to the existing `shift` and `fillna` ops, but I didn't think there was a clear way to pass in an anonymous func to `post_processing` that would have access to the sliced value that that function needs.

Reply:
ok for now. these should be subclasses of common routines (in a separate module), then this will be much simpler / cleaner (but that's for later)